src/lib/Sympa/Message.pm

```diff
@@ -2098,8 +2098,15 @@ sub _urlize_parts {
         return undef;
     }
 
-    ## Clean up Message-ID
-    my $dir1 = Sympa::Tools::Text::escape_chars($message_id);
+    ## Clean up Message-ID and preventing double percent encoding.
```

Comment:
encode_filesystem_safe() would be better to be used instead of ad-hoc escaping.

Reply:
Indeed. I never noticed this function.
Fixed in commit #c8195a6